src/web/components/CommentsLayout.tsx

```diff
@@ -56,49 +61,123 @@ export const CommentsLayout = ({
     discussionApiClientHeader,
     commentToScrollTo,
     onPermalinkClick,
-}: Props) => (
-    <Flex direction="row">
-        <LeftColumn showRightBorder={false}>
-            <SignedInAs
-                pillar={pillar}
-                enableDiscussionSwitch={enableDiscussionSwitch}
-                user={user}
-                commentCount={commentCount}
-                isClosedForComments={isClosedForComments}
-            />
-        </LeftColumn>
-        <div className={containerStyles}>
-            <Hide when="above" breakpoint="leftCol">
-                <div className={bottomPadding}>
-                    <SignedInAs
-                        pillar={pillar}
-                        enableDiscussionSwitch={enableDiscussionSwitch}
+}: Props) => {
+    const [heightToStick, setHeightToStick] = useState(DEFAULT_HEIGHT);
+    const [heightChangeCount, setheightChangeCount] = useState<number>(0);
+    const heightRef = useRef<HTMLDivElement>(null);
+
+    const handleHeightChange = () => {
+        setheightChangeCount(heightChangeCount + 1);
+    };
+
+    const checkHeight = (ref: RefObject<HTMLDivElement>) => {
+        if (ref.current) {
+            const availableHeight = ref.current.clientHeight;
+            // Don't run the sticky container all the way to the bottom
+            let heightWereUsing = availableHeight - 600;
+            // Never try to use a height less than the default size
+            if (heightWereUsing < DEFAULT_HEIGHT) {
+                heightWereUsing = DEFAULT_HEIGHT;
+            }
+            setHeightToStick(heightWereUsing);
+        }
+    };
+
+    useEffect(() => {
+        checkHeight(heightRef);
+    }, [heightChangeCount]);
+
+    useEffect(() => {
+        // Whoah, why on earth are you doing this? Are you mad?
+        // Yup, this is a hackity, hack, hack. It's bad because we have no way to be sure that the dom
+        // has finished rendering so maybe the check will fail (return the wrong height), and it's also
+        // bad because we could be running the check too late (delaying the dom update). So why?
+        // Because we are already working on a better way to render this ad slot so we're happy
+        // to leave this code sitting here for now as a workaround
+        // TODO: **** Remove this ****
+        if (expanded) {
+            setTimeout(() => {
+                checkHeight(heightRef);
+            }, 500);
+            setTimeout(() => {
+                checkHeight(heightRef);
+            }, 3000);
```

Comment:
As it's a hack, maybe not worth it but `setInterval` might respond quicker?
```suggestion
			let intervalCount = 0;
            let checkHeightInterval = setInterval(() => {
	            intervalCount++;
                checkHeight(heightRef);
    			if (intervalCount === 8) {
    				clearInterval(checkHeightInterval)
    			}
            }, 500);
```